gnovm/pkg/gnolang/machine.go

```diff
@@ -402,9 +402,15 @@ func (m *Machine) runFiles(fns ...*FileNode) {
 	// Files' package names must match the machine's active one.
 	// if there is one.
 	for _, fn := range fns {
-		if fn.PkgName != "" && fn.PkgName != m.Package.PkgName {
-			panic(fmt.Sprintf("expected package name [%s] but got [%s]",
-				m.Package.PkgName, fn.PkgName))
+		var (
+			notEmptyName = fn.PkgName != ""
+			notBaseName  = fn.PkgName != m.Package.PkgName
+			notTestName  = m.Package.PkgName+"_test" != fn.PkgName
+		)
+
+		if notEmptyName && notBaseName && notTestName {
```

Comment:
I'm not 100% sure this is correct. Is a better approach here not that only files with a same package name are accepted by RunFiles, and cmd/gno/test is responsible for doing two dfferent RunFiles?

My reasoning is that if we are running RunFiles on `pkg_test`, then we should not even consider the normal files with package `pkg`. If anything, package `pkg_test` _may_ then decide to import `pkg` and execute it.

So I would say it is better to only pass the `pkg_test` subset of files or the `pkg` subset of files?

Reply:
Just for reference, this is where I think the current code might fail:

```
// a.gno
package a

var X = 1

///

// a_test.gno
package a_test

import "gno.land/r/demo/a"

var X = 2

func (t *testing.T) TestX(t *testing.T) { t.Log(X) } // should output 2
```

Not entirely sure, might need to do another review.